### PR TITLE
Reorganize the ACA-Py Documentation Files

### PR DESCRIPTION
**Pull request by swcurran**
This PR does a radical reorganization of the documents in the ACA-Py repository. The following is a summary of the changes:

- Most Markdown files in the root have been moved to the docs folder, and organized into folders that match those used on the [https://aca-py.org](https://aca-py.org) -- demo, gettingStarted, testing, deploying, etc.
- Expected files -- LICENSE, MAINTAINERS.md, README.md and the like were left in the root folder.
- The MD files in the `demo` folder were moved to `docs/demo' and a new README.md file added pointing to those.
- The previous Getting Started files were moved to the `docs/gettingStarted` folders.
- A pass was done to change all the links found (in a manual pass...) to be relative links and to the new location of the files.
- A pass was done to remove all of the MD lint warnings (other than spelling of ACA-Py/Aries/SSI terms).
- Some cleanup was done to external references -- changing references from greenlight ledger to BCovrin test, eliminating some references to old demos, adding references to the Traction Tutorial.
- New README.md files were needed -- such as in the root of the `docs` folder with pointers to guidance for maintaining both the ReadTheDocs documentation and the [https://aca-py.org](https://aca-py.org) documentation.
- I've not done anything with the ELK Stack MD documents -- will think about what to do with those later (if anything).

Not perfect, but I think it is close.

It is a lot of change. Would appreciate people browsing [my branch](https://github.com/swcurran/aries-cloudagent-python/tree/organize-docs) in the GitHub UI to scan for issues, and report them.  Important things to note are changes that need to be made before we merge this and more general updates that are overdue and needed.

### CODE REVIEW

**Comment by swcurran** [comment]
Not good that we have an integration test error when moving docs :-).

The error is from this test: ` features/0586-sign-transaction.feature:204  endorse a schema and cred def transaction, write to the ledger, issue and revoke a credential, with auto endorsing workflow -- @1.2 `

Not sure if it is a timing issue or something else. Seems to be finding duplicate records. Perhaps someone could run this locally.  It is related to the Endorser, so perhaps updates in that area?

File: README.md [hunk 1]
```diff
@@ -8,38 +8,41 @@
...
+ACA-Py supports a Transaction Endorsement protocol, for agents that don't have write access to an Indy ledger.  Endorser support is documented [here](docs/features/Endorser.md).
 
 ### Scaled Deployments
 
 ACA-Py supports deployments in scaled environments such as in Kubernetes environments where ACA-Py and its storage components can be horizontally scaled as needed to handle the load.
 
 ### VC-API Endpoints
 
-A set of endpoints conforming to the vc-api specification are included to manage w3c credentials and presentations. They are documented [here](JsonLdCredentials.md#vc-api) and a postman demo is available [here](./demo/AriesPostmanDemo.md#vc-api).
+A set of endpoints conforming to the vc-api specification are included to manage w3c credentials and presentations. They are documented [here](docs/features/JsonLdCredentials.md#vc-api) and a postman demo is available [here](docs/demos/AriesPostmanDemo.md#vc-api).
```
Comment (jamshale) [inline README.md:45]: Wrong path here. Should be `docs/demo/AriesPostmanDemo`

File: PUBLISHING.md [hunk 2]
```diff
@@ -145,11 +145,10 @@ Once you have the list of PRs:
    "Packages").
 
    Additional information about the container image publication process can be
-   found in the document [Container Images and Github Actions]().
+   found in the document [Container Images and Github Actions](../deploying/ContainerImagesAndGithubActions.md).
```
Comment (jamshale) [inline PUBLISHING.md:148]: Think this link is broken.

**Comment by jamshale** [comment]
> Not sure if it is a timing issue or something else. Seems to be finding duplicate records. Perhaps someone could run this locally. It is related to the Endorser, so perhaps updates in that area?

I'm pretty sure this is an intermittently failing test, as other tests off of main are passing and I think I've seen it before. Re-running. If it fails again I'll look at it locally.

**Review by dbluhm — COMMENTED** [review]
LGTM! Truthfully, I didn't read through every change but I'll rely on @jamshale's more in depth review and the linting to catch issues with links and such.

**Review by dbluhm — APPROVED** [review]
Meant to hit approve...

**Comment by ianco** [comment]
> > Not sure if it is a timing issue or something else. Seems to be finding duplicate records. Perhaps someone could run this locally. It is related to the Endorser, so perhaps updates in that area?
> 
> I'm pretty sure this is an intermittently failing test, as other tests off of main are passing and I think I've seen it before. Re-running. If it fails again I'll look at it locally.

I've noticed the occasional timing issue with the ledger.  I.e. the test registers a DID and then when it tries to record a schema it get an error that the DID isn't approved to update the ledger.  I think it's a node synchronization timing issue.

**Review by ianco — APPROVED** [review]
Overall looks good.  (Like Daniel I didn't read through every single document.)  My preference would be to leave the demo docs in the `/demo` folder, but I don't have a strong opinion about it.

**Comment by swcurran** [comment]
I can’t believe y’all didn’t read every change in every file!!!  IMHO, we want to encourage people to use https://aca-py.org, so having a redirector page in the /demo folder is the right thing to do.  Understand your concerns, but at least they are all in one place — /docs/demo.

Thanks for the reviews!

**Pull request merged**